### PR TITLE
fix storage in default scenario helptext

### PR DESCRIPTION
**Pull request by Derryrover**

### CODE REVIEW

File: src/reducers.js [hunk 1]
```diff
@@ -100,6 +102,29 @@ function organisations(
...
+function usage (
+  state = {
+    raster_count: 0,
+    raster_total_size: 0,
+    scenario_count: 0,
+    scenario_total_size: 0,
+    timeseries_count: 0,
+    timeseries_total_size: 0,
+    isFetching: false,
+    timesFetched: 0,
```
Comment (hoanphungt) [inline src/reducers.js:114]: Is there a reason to use both snake cases and camel cases here? I guess because data comes from API is in snake case.

File: src/data_management/rasters/RasterSourceTable.tsx [hunk 2]
```diff
@@ -25,6 +29,8 @@ export const RasterSourceTable = (props:any) =>  {
   const [deleteFunction, setDeleteFunction] = useState<null | Function>(null);
   const [busyDeleting, setBusyDeleting] = useState<boolean>(false);
   const [showDeleteFailedModal, setShowDeleteFailedModal] = useState<boolean>(false);
+  const scenarioTotalSize = useSelector(getScenarioTotalSize);
```
Comment (hoanphungt) [inline src/data_management/rasters/RasterSourceTable.tsx:32]: I think you should get `rasterTotalSize` here instead of `scenarioTotalSize`

File: src/actions.js [hunk 3]
```diff
@@ -86,7 +87,7 @@ export function fetchOrganisations() {
     fetch(url, opts)
       .then(responseObj => responseObj.json())
       .then(responseData => {
-        const organisation = state.organisations.selected;
+        const selectedOrganisationLocalStorage = getLocalStorage("lizard-management-current-organisation", null);
```
Comment (hoanphungt) [inline src/actions.js:90]: I don't get why you need to get the selected organisation from the local storage and how is this change related to this issue to show the storage of rasters and scenarios. Does it fix have anything to do with this bug: https://github.com/nens/lizard-management-client/issues/680?

File: src/components/ExplainSideColumn.tsx [hunk 4]
```diff
@@ -11,6 +11,23 @@ interface MyProps {
...
+    if (typeof value === "object" && value !== null) {
+      if (seen.has(value)) {
+        return;
+      }
+      seen.add(value);
+    }
+    return value;
+  };
+};
+
```
Comment (hoanphungt) [inline src/components/ExplainSideColumn.tsx:30]: I think this is because you are currently saving the selected organisation in the local storage as a string and then later on, you try to stringify this string again and it caused a circle. I think it could be solved by not storing the selected organisation in local storage as JSON string but as an object.

Comment (hoanphungt) [reply src/components/ExplainSideColumn.tsx:30]: I just checked and it's not possible to store an object in local storage. So it means you need to parse it to an object first before using it.

File: src/actions.js (hunk 3 above)
Comment (hoanphungt) [reply src/actions.js:90]: Also, this leads to bug. If I don't have any selected organisation in the local storage, it will cause error: `Cannot read property 'roles' of null`. So overall, I think this is not a good change.

**Review by hoanphungt — COMMENTED** [review]
Comments in PR

File: src/components/ExplainSideColumn.tsx (hunk 4 above)
Comment (Derryrover) [reply src/components/ExplainSideColumn.tsx:30]: It is needed for the line below that uses a jsx component as key:
key={JSON.stringify(explanationText, getCircularReplacer)}
We could also adapt the JSX element manually to a different non circular object.
Anyway I suspect that introduces new surface for bugs, so I prefer this patch.

File: src/data_management/rasters/RasterSourceTable.tsx (hunk 2 above)
Comment (Derryrover) [reply src/data_management/rasters/RasterSourceTable.tsx:32]: fixed

File: src/reducers.js (hunk 1 above)
Comment (Derryrover) [reply src/reducers.js:114]: Yes because it comes from the api. I choose this approach of changing as least as possible to the original data structure.

File: src/actions.js (hunk 3 above)
Comment (Derryrover) [reply src/actions.js:90]: You are right!
I need to look into this.
I will start to merge the fix for the 680 bug to see if it helps.

File: src/reducers.js [hunk 5]
```diff
@@ -60,7 +61,7 @@ function organisations(
     timesFetched: 0,
     available: [],
     availableForRasterSharedWith: [],
-    selected: getLocalStorage("lizard-management-current-organisation", null),
+    selected: null,
```
Comment (Derryrover) [inline src/reducers.js:64]: I could remove "getLocalStorage("lizard-management-current-organisation" here because I dispatch it in the action instead.
Dispatching it in the action allows me to also trigger a fetch to the usage api on startup.

File: src/actions.js (hunk 3 above)
Comment (Derryrover) [reply src/actions.js:90]: @hoanphungt 
Hi Hoan,
I reproduced your error and believe I fixed it.
Also in the meanwhile I found bugs with the routing that I also fixed (I think so).
The "getLocalStorage("lizard-management-current-organisation", null); " here is needed to retrieve the selected organisation that the user selected during last usage of the app.
Previously this was done in the reducer default value, but here it is easier to also trigger the 'usage' api on startup. This usage api is fetched anytime a selected organisation is changed through the action. (but not if it is the default of the redux store).

**Review by hoanphungt — APPROVED** [review]
I think it looks good but needs to be tested on staging to see if these changes work correctly. Anyway, I think it would be better if you fix the routing issue and selected organisation bug in a separate issue/PR.

One more point, when I test it, the animation effect of the explain box is gone (so the border color of the explain box is not changing anymore when switching to another field). I will make a commit to improve this part a bit.
